File: src/_1ProgrammingBasicsWithJavaMay2023/_3ConditionalStatementsAdvanced/_3MoreExercise/_05Vacation.java
package _1ProgrammingBasicsWithJavaMay2023._3ConditionalStatementsAdvanced._3MoreExercise;

import java.util.Scanner;

public class _05Vacation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double budget = Double.parseDouble(scanner.nextLine());
        String season = scanner.nextLine();
        String place = "";
        String location = "";
        double price = 0.00;
        if (budget <= 1000) {
            place = "Camp";
            if (season.equals("Summer")) {
                location = "Alaska";
                price = budget * 0.65;
            } else if (season.equals("Winter")) {
                location = "Morocco";
                price = budget * 0.45;
            }
        } else if (budget > 1000 && budget <= 3000) {
            place = "Hut";
            if (season.equals("Summer")) {
                location = "Alaska";
                price = budget * 0.80;
            } else if (season.equals("Winter")) {
                location = "Morocco";
                price = budget * 0.60;
            }
        } else if (budget > 3000) {
            place = "Hotel";
            if (season.equals("Summer")) {
                location = "Alaska";
                price = budget * 0.90;
            } else if (season.equals("Winter")) {
                location = "Morocco";
                price = budget * 0.90;
            }
        }
        System.out.printf("%s - %s - %.2f", location, place, price);
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_3ConditionalStatementsAdvanced/_3MoreExercise/_05Vacation.js
function vacation(budget, season) {
    const ALASKA_CAMP_PRICE_PERCENT = 0.65;
    const ALASKA_HUT_PRICE_PERCENT = 0.80;
    const ALASKA_HOTEL_PRICE_PERCENT = 0.90;
    const MOROCCO_CAMP_PRICE_PERCENT = 0.45;
    const MOROCCO_HUT_PRICE_PERCENT = 0.60;
    const MOROCCO_HOTEL_PRICE_PERCENT = 0.90;

    let placeToStay;
    let countryToStay;
    let priceToStay = 0.00;

    if (season === "Summer") {
        countryToStay = "Alaska";
        if (budget <= 1000) {
            placeToStay = "Camp";
            priceToStay = budget * ALASKA_CAMP_PRICE_PERCENT;
        } else if (budget <= 3000) {
            placeToStay = "Hut";
            priceToStay = budget * ALASKA_HUT_PRICE_PERCENT;
        } else if (budget > 3000) {
            placeToStay = "Hotel";
            priceToStay = budget * ALASKA_HOTEL_PRICE_PERCENT;
        }
    } else if (season === "Winter") {
        countryToStay = "Morocco";
        if (budget <= 1000) {
            placeToStay = "Camp";
            priceToStay = budget * MOROCCO_CAMP_PRICE_PERCENT;
        } else if (budget <= 3000) {
            placeToStay = "Hut";
            priceToStay = budget * MOROCCO_HUT_PRICE_PERCENT;
        } else if (budget > 3000) {
            placeToStay = "Hotel";
            priceToStay = budget * MOROCCO_HOTEL_PRICE_PERCENT;
        }
    }

    console.log(`${countryToStay} - ${placeToStay} - ${priceToStay.toFixed(2)}`);
}